cheroot/server.py

```diff
@@ -1155,6 +1156,28 @@ def close(self):
             # Apache does, but not today.
             pass
 
+    def get_peer_uid(self):
+        """Returns the UID of the peer socket in the case of UNIX domain sockets
+
+        This function uses SO_PEERCRED to query the UNIX UID of the peer, which
+        is only available if the bind address is a UNIX domain socket.
+        """
+
+        if self.socket.family != socket.AF_UNIX:
+            return None
+
+        # NOTE: the value for SO_PEERCRED can be architecture specific, in
+        # which case the getsockopt() will hopefully fail.  The arch
+        # specific value could be derived from platform.processor()
+        SO_PEERCRED = getattr(socket, 'SO_PEERCRED', 17)
+        try:
+            creds = self.socket.getsockopt(socket.SOL_SOCKET, SO_PEERCRED, struct.calcsize('3i'))
+            pid, uid, gid = struct.unpack('3i', creds)
+            return uid
+        except socket.error:
+            pass
```

Comment:
Alright, warning would be ok (mabe even  `NOTICE`). But I still prefer raising exception instead of returning `None`.